core/src/style/style.cpp

```diff
@@ -69,10 +79,59 @@ void Style::setLightingType(LightingType _type){
 
 }
 
-void Style::addLayer(const std::pair<std::string, StyleParamMap>&& _layer) {
+void Style::addLayer(SceneLayer* _layer) {
+
+    m_layers.push_back(_layer);
+
+}
+
+void Style::applyLayerFiltering(const Feature& _feature, const Context& _ctx, long long& _uniqueID,
+                                   StyleParamMap& _styleParamMapMix, SceneLayer* _uberLayer) const {
+
+    std::vector<SceneLayer*> sLayers;
+    sLayers.reserve(_uberLayer->getSublayers().size() + 1);
+    sLayers.push_back(_uberLayer);
+
+    auto sLayerItr = sLayers.begin();
+
+    //A BFS traversal of the SceneLayer graph
+    while (sLayerItr != sLayers.end()) {
+
+        if ( (*sLayerItr)->getFilter()->eval(_feature, _ctx)) { // filter matches
+
+            _uniqueID += (1 << (*sLayerItr)->getID());
```

Comment:
Using bitmasks would be a really efficient way of indexing style combinations, but as you mention below it restricts the number of possible sublayers and that's not gonna be acceptable in the long run. Also, I'm not sure this will behave correctly in every case; is style parameter mixing order-dependent?